---
title: "Class 15: Pertussis and CMI-PB mini project"
author: "Morgan Black (PID A14904860)"
format: pdf
---

The CDC tracks pertussis case numbers which can be accessed through their website [here](https://www.cdc.gov/pertussis/php/surveillance/pertussis-cases-by-year.html?CDC_AAref_Val=https://www.cdc.gov/pertussis/surv-reporting/cases-by-year.html)

First we need to "scrape" this data to import it into R using the **datapasta** package.
```{r}

cdc <- data.frame(
  year = c(
    1922L, 1923L, 1924L, 1925L,
    1926L, 1927L, 1928L, 1929L, 1930L, 1931L,
    1932L, 1933L, 1934L, 1935L, 1936L,
    1937L, 1938L, 1939L, 1940L, 1941L, 1942L,
    1943L, 1944L, 1945L, 1946L, 1947L,
    1948L, 1949L, 1950L, 1951L, 1952L,
    1953L, 1954L, 1955L, 1956L, 1957L, 1958L,
    1959L, 1960L, 1961L, 1962L, 1963L,
    1964L, 1965L, 1966L, 1967L, 1968L, 1969L,
    1970L, 1971L, 1972L, 1973L, 1974L,
    1975L, 1976L, 1977L, 1978L, 1979L, 1980L,
    1981L, 1982L, 1983L, 1984L, 1985L,
    1986L, 1987L, 1988L, 1989L, 1990L,
    1991L, 1992L, 1993L, 1994L, 1995L, 1996L,
    1997L, 1998L, 1999L, 2000L, 2001L,
    2002L, 2003L, 2004L, 2005L, 2006L, 2007L,
    2008L, 2009L, 2010L, 2011L, 2012L,
    2013L, 2014L, 2015L, 2016L, 2017L, 2018L,
    2019L, 2020L, 2021L, 2022L, 2024L
  ),
  cases = c(
    107473, 164191, 165418, 152003,
    202210, 181411, 161799, 197371,
    166914, 172559, 215343, 179135, 265269,
    180518, 147237, 214652, 227319, 103188,
    183866, 222202, 191383, 191890, 109873,
    133792, 109860, 156517, 74715, 69479,
    120718, 68687, 45030, 37129, 60886,
    62786, 31732, 28295, 32148, 40005,
    14809, 11468, 17749, 17135, 13005, 6799,
    7717, 9718, 4810, 3285, 4249, 3036,
    3287, 1759, 2402, 1738, 1010, 2177, 2063,
    1623, 1730, 1248, 1895, 2463, 2276,
    3589, 4195, 2823, 3450, 4157, 4570,
    2719, 4083, 6586, 4617, 5137, 7796, 6564,
    7405, 7298, 7867, 7580, 9771, 11647,
    25827, 25616, 15632, 10454, 13278,
    16858, 27550, 18719, 48277, 28639, 32971,
    20762, 17972, 18975, 15609, 18617,
    6124, 2116, 3044, 23544
  )
)

```


```{r}

library(ggplot2)
baseplot <- ggplot(cdc) + aes(year, cases) +
              geom_point() +
              geom_line()
               
baseplot

```

Now let's add the data of vaccine rollout.
```{r}

baseplot +
  theme_bw() +
  geom_vline(xintercept=1946, col="blue") + #When the wP vaccine came out
  geom_vline(xintercept=1996, col="red") + #Switch from wP to the new aP vaccine
  geom_vline(xintercept=2020, col="green") + #COVID pandemic
  geom_vline(xintercept=2003, col="gray") #first big uptick after the new vaccine, seems to confer resistance for less time than the wP vaccine

```



## CMI_PB (Computational Models of Immunity - Pertussis Boost)

This project collects and makes freely available data about the immune response to pertussis vaccination. Access the data via an API which returns JSON format. 
Use **jsonlite** package to read in the data:

```{r}

library(jsonlite)

subject <- read_json("https://www.cmi-pb.org/api/v5/subject", 
                     simplifyVector = TRUE)

head(subject)

```

### Q: How many subjects are in this dataset?
```{r}

nrow(subject)

```

### Q: How many male/female subjects are there?
```{r}

table(subject$biological_sex)

```


### Q: How many wP and aP do we have?
```{r}

table(subject$infancy_vac)

```


### Q: Breakdown of biological sex and race?
```{r}

table(subject$race, subject$biological_sex)

```

### Q: Does this breakdown relfect the US population?
No!
 
The dataset is still large and useful, so although it will not reflect the population, it's still worth looking at.


```{r}

table(subject$dataset)

```


```{r}

specimen <- read_json("http://cmi-pb.org/api/v5/specimen", 
                      simplifyVector = TRUE)
ab_titer <- read_json("http://cmi-pb.org/api/v5/plasma_ab_titer", 
                      simplifyVector = TRUE)

```


We want to merge or "join" these tables so we can have all the info we need for antibody measurements
```{r}

library(dplyr)

meta <- inner_join(subject, specimen)

```
```{r}

head(meta)

```


Now we join 'ab_titer' and 'meta'
```{r}

abdata <- inner_join(ab_titer, meta)

```

```{r}

nrow(abdata)

```

```{r}

table(abdata$isotype)

```

```{r}

table(abdata$antigen)

```


Let's begin by looking at IgG

```{r}

igg <- filter(abdata, isotype == "IgG")
head(igg)

```
Make a boxplot of IgG antigen levels- this will be a plot of MFI vs antigen
```{r}

ggplot(igg) +
  aes(MFI_normalised, antigen) +
  geom_boxplot()

```
Color by infancy vaccine:
```{r}

ggplot(igg) +
  aes(MFI_normalised, antigen, col=infancy_vac) +
  geom_boxplot()

```



Ideally we'd like to see how these levels change over time relative to the booster shot dose time

```{r}

table(abdata$visit)

```

```{r}

ggplot(igg) +
  aes(MFI_normalised, antigen) +
  geom_boxplot() +
  facet_wrap(~infancy_vac)

```

```{r}

ggplot(igg) +
  aes(MFI_normalised, antigen) +
  geom_boxplot() +
  facet_wrap(~visit)

```

```{r}

ggplot(igg) +
  aes(MFI_normalised, antigen, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(~visit)


```


```{r}

abdata_2021 <- abdata %>% filter(dataset == "2021_dataset")

abdata_2021 %>% filter(isotype == "IgG", antigen == "PT") %>%
  ggplot() +
  aes(planned_day_relative_to_boost, MFI_normalised, col= infancy_vac, group=subject_id) +
  geom_point() +
  geom_line() +
  geom_vline(xintercept=0, linetype="dashed") +
  geom_vline(xintercept=14, linetype="dashed") +
  labs(title="2021 dataset IgG PT",
       subtitle = "Day 0 (pre-booster) and 14 (apparent peak levels) marked by dashed lines") 
  

```

Let's look at the same data from 2020 and compare plots:
```{r}

abdata_2020 <- abdata %>% filter(dataset == "2020_dataset")

abdata_2020 %>% filter(isotype == "IgG", antigen == "PT") %>%
  ggplot() +
  aes(planned_day_relative_to_boost, MFI_normalised, col= infancy_vac, group=subject_id) +
  geom_point() +
  geom_line() +
  geom_vline(xintercept=0, linetype="dashed") +
  geom_vline(xintercept=14, linetype="dashed") +
  labs(title="2020 dataset IgG PT",
       subtitle = "Day 0 (pre-booster) and 14 (apparent peak levels) marked by dashed lines") 

```
The 2020 dataset seems to have been interrupted by the pandemic with some patients coming back for their final collection point far after other patients. However the dynamics between day 0 to day 120 seems similar to the 2021 dataset, other than the two pink lines (two patients that initially got the aP vaccine) that have large upward/downward spikes in normalized MFI compared to the other patients. 


## CMI-PB RNAseq Data

```{r}

url <- "https://www.cmi-pb.org/api/v2/rnaseq?versioned_ensembl_gene_id=eq.ENSG00000211896.7"

rna <- read_json(url, simplifyVector = TRUE) 

```

```{r}

ssrna <- inner_join(rna, meta)

```
Plot time course of gene expression for the IGHG1 gene

```{r}

ggplot(ssrna) +
  aes(visit, tpm, group=subject_id) +
  geom_point() +
  geom_line(alpha=0.2)

```
It's hard to tell individual subjects by this plot, but it seems that the peak expression hits for most patients around visit 4 after the booster shot. If this corresponds to days, then this is about a week earlier than the peak IgG antigen levels seen in the previous plots, but aligns around the point where antigen levels start to increase for each patient. If this doesn't correspond to days post-booster, then it's hard to tell how this correlates time-wise with the antigen tracking plot above. 


Color by infancy_vac:
```{r}

ggplot(ssrna) +
  aes(tpm, col=infancy_vac) +
  geom_boxplot() +
  facet_wrap(vars(visit))

```

Focus in on a particular visit (visit 4) to see if there are differences between aP and wP:
```{r}

ssrna %>%  
  filter(visit==4) %>% 
  ggplot() +
    aes(tpm, col=infancy_vac) + geom_density() + 
    geom_rug() 

```





